apps/mobile-wallet/src/components/NFTImage.tsx

```diff
@@ -12,23 +13,51 @@ import { BORDER_RADIUS_SMALL } from '~/style/globalStyle'
 export interface NFTImageProps {
   nftId: NFT['id']
   size?: DimensionValue
+  play?: boolean
 }
 
-const NFTImage = ({ nftId, size = '100%' }: NFTImageProps) => {
+enum NFTDataTypes {
+  image = 'image',
+  video = 'video',
+  audio = 'audio',
+  other = 'other'
+}
+
+type NFTDataType = keyof typeof NFTDataTypes
+
+const NFTImage = memo(({ nftId, size = '100%', play }: NFTImageProps) => {
   const nft = useAppSelector((s) => selectNFTById(s, nftId))
   const theme = useTheme()
 
   const [hasError, setHasError] = useState(false)
+  const [contentType, setContentType] = useState<NFTDataType>()
+
+  useEffect(() => {
+    if (nft?.image)
+      fetch(nft.image).then((res) => {
+        const contentType = res.headers.get('content-type') || ''
+        const contentTypeCategory = contentType.split('/')[0]
+
+        setContentType(contentTypeCategory in NFTDataTypes ? (contentTypeCategory as NFTDataType) : 'other')
+      })
+  }, [nft?.image])
 
   if (!nft) return null
 
   const isDataUri = nft.image.startsWith('data:image/')
 
-  return hasError ? (
-    <NFTPlaceholder size={size} />
-  ) : isDataUri ? (
-    <NFTWebView imageUri={nft.image} size={size} />
-  ) : (
+  if (!contentType) return <NFTPlaceholder size={size} showCameraOff={false} />
+
+  // Loading many videos at once is too heavy on Android.
+  // Using the image component for NFT lists on Android is sufficient.
+  if (contentType === 'video' && (Platform.OS === 'ios' || play))
```

Comment:
Not sure this conditon is correct because the `Platform.OS === 'ios'` has no effect (because of the use of `||`)
- on iOS, this will auto play if `play` is true
- on Android, this will also auto play if `play` is true

As it is today, we can remove the `Platform.OS === 'ios'` as you use the props properly (we don't autoplay in the NFTGrid for both platform which is fine by me).
If you wanted to add a safeguard to avoid wastful autoplay on Android, we may need 2 different props (`play` & `autoPlay`).

Reply:
As the comment above describes, we use the `NftVideo` component when:
1. `contentType === 'video' && Platform.OS === 'ios'`
2. `contentType === 'video' && play`

Reply:
I changed the condition to make it a bit clearer, does it help?

Reply:
Ah I didn't understand that even just loading the video component was already too much!
Thanks for answering, clearer now